microceph/api/cluster.go

```diff
@@ -0,0 +1,73 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/ceph"
+	"github.com/canonical/microceph/microceph/constants"
+	"github.com/canonical/microceph/microceph/interfaces"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+)
+
+var clusterCmd = rest.Endpoint{
+	Path: "cluster",
+	Get:  rest.EndpointAction{Handler: cmdClusterGet, ProxyTarget: false},
+}
+
+// cmdClusterGet returns a json dump of microceph configs with a new key patched in for the remote.
+var cmdClusterGet = func(s *state.State, r *http.Request) response.Response {
+	// Fetch request params.
+	var req types.ClusterStateRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.InternalError(err)
+	}
+
+	// Check that the cluster name is conformant.
+	isOk, err := regexp.MatchString(constants.ClusterNameRegex, req.RemoteName)
+	if err != nil || !isOk {
+		err := fmt.Errorf("cluster names can only have [a-z] or [0-9] characters: %w", err)
+		logger.Error(err.Error())
+		return response.InternalError(err)
```

Comment:
This should be a `BadRequest`, since `InternalError` doesn't represent a user error, imo.